### ISSUE TITLE
Compilation error in generated UDTRecord's POJO constructor if the UDT contains arrays of table records

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13315]
A test case added for https://github.com/jOOQ/jOOQ/issues/8614 looks like this:

```sql
CREATE TABLE t_8614 (i int, j int);

CREATE TYPE u_8614 AS (
  r t_8614[]
);
```

The generated `U_8614Record` has a POJO constructor like this:

```java
    /**
     * Create a detached, initialised U_8614Record
     */
    public U_8614Record(org.jooq.test.postgres.generatedclasses.public_.udt.pojos.U_8614 value) {
        super(U_8614.U_8614);

        if (value != null) {
            setR(value.getR());
        }
    }
```

There's a compilation error, because `setR(T_8614Record...)` is incompatible with `getR():T_8614[]`.

This only happens with UDTs and POJOs enabled, and when the array is an array of records.

**Comment by lukaseder** [comment]
The problem is the array of table records. An array of UDT records is already supported, so it's really quite the edge case.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.6 (https://github.com/jOOQ/jOOQ/issues/13316) and 3.15.10 (https://github.com/jOOQ/jOOQ/issues/13317)
